### DISCUSSION TITLE
[item]MeteData table

### DISCUSSION THREAD

**Discussion by Wang-Xiaofei** [windingwind/zotero-better-notes#401] [Note Templates]
<table>    
 <tr>         
	<th style="background-color:#dbeedd;">              
	     <p style="text-align: right">Title: </p>          
	</th>         
	<td style="background-color:#dbeedd;">${topItem.getField('title')}         
	</td>      
 </tr> 
 <tr>
        <th style="background-color:#f3faf4;">             
	   <p style="text-align: right">Journal: </p>        
       </th>         
       <td style="background-color:#f3faf4;">${topItem.getField('publicationTitle')}   
       </td>    
 </tr>     
 <tr>
        <th style="background-color:#f3faf4;">
            
            <p style="text-align: right">Authors: </p>
        </th>
        <td style="background-color:#f3faf4;">${topItem.getCreators().map((v)=>v.lastName+" "+v.firstName).join("; ")}
        </td>
 </tr>
 <tr>
        <th style="background-color:#dbeedd;">   
            <p style="text-align: right">Pub.date: </p>
        </th>
        <td style="background-color:#dbeedd;">${topItem.getField('date')}
        </td>
 </tr>   
 <tr>
        <th style="background-color:#f3faf4;">
            <p style="text-align: right">DOI: </p>
        </th>
        <td style="background-color:#f3faf4;">
            <a href= "https://doi.org/${topItem.getField('DOI')}">${topItem.getField('DOI')}</a>
        </td>
 </tr>
 <tr>
        <th style="background-color:#dbeedd;">
            
            <p style="text-align: right">期刊标签: </p>
        </th>
        <td style="background-color:#dbeedd;">${topItem.getField('seriesText')}
        </td>
 </tr>
  <tr>
        <th style="background-color:#dbeedd;">
            
            <p style="text-align: right">分区: </p>
        </th>
        <td style="background-color:#dbeedd;">${topItem.getField('seriesTitle')}
        </td>
 </tr>
 <tr>
        <th style="background-color:#dbeedd;">
             
            <p style="text-align: right">IF: </p>
        </th>
        <td style="background-color:#dbeedd;">${topItem.getField('callNumber')}
        </td>
 </tr>
 </table>

大概效果如下：
![image](https://user-images.githubusercontent.com/52536738/227159659-62212c4c-29b6-4293-b1c7-b6cea511f560.png)

**Comment by Wang-Xiaofei** [comment]
```HTML

<table>    
 <tr>         
	<th style="background-color:#dbeedd;">              
	     <p style="text-align: right">Title: </p>          <!-- 标题 -->
	</th>         
	<td style="background-color:#dbeedd;">${topItem.getField('title')}         
	</td>      
 </tr> 
 <tr>
        <th style="background-color:#f3faf4;">             
	   <p style="text-align: right">Journal: </p>        <!-- 期刊  -->
       </th>         
       <td style="background-color:#f3faf4;">${topItem.getField('publicationTitle')}   
       </td>    
 </tr>     
 <tr>
        <th style="background-color:#f3faf4;">
            <!-- 作者 -->
            <p style="text-align: right">Authors: </p>
        </th>
        <td style="background-color:#f3faf4;">${topItem.getCreators().map((v)=>v.lastName+" "+v.firstName).join("; ")}
        </td>
 </tr>
 <tr>
        <th style="background-color:#dbeedd;">   <!-- 出版日期 -->
            <p style="text-align: right">Pub.date: </p>
        </th>
        <td style="background-color:#dbeedd;">${topItem.getField('date')}
        </td>
 </tr>   
 <tr>
        <th style="background-color:#f3faf4;">
            <p style="text-align: right">DOI: </p>
        </th>
        <td style="background-color:#f3faf4;">
            <a href= "https://doi.org/${topItem.getField('DOI')}">${topItem.getField('DOI')}</a>
        </td>
 </tr>
 <tr>
        <th style="background-color:#dbeedd;">
            <!-- 期刊标签 -->
            <p style="text-align: right">期刊标签: </p>
        </th>
        <td style="background-color:#dbeedd;">${topItem.getField('seriesText')}
        </td>
 </tr>
  <tr>
        <th style="background-color:#dbeedd;">
            <!-- 分区 -->
            <p style="text-align: right">分区: </p>
        </th>
        <td style="background-color:#dbeedd;">${topItem.getField('seriesTitle')}
        </td>
 </tr>
 <tr>
        <th style="background-color:#dbeedd;">
             <!-- 影响因子 -->
            <p style="text-align: right">IF: </p>
        </th>
        <td style="background-color:#dbeedd;">${topItem.getField('callNumber')}
        </td>
 </tr>
 </table>

```

**Comment by EscapeQuickly** [comment]
不能运行呀

**Reply by Wang-Xiaofei** [reply]
上个可能有些问题，我更新了新的模板

**Comment by Wang-Xiaofei** [comment]
现在模板更新了，以下这个模板可以用

```HTML
<h1>${topItem.getField("title")}</h1>   <!-- author:cople -->
<h2 style="color: #1E5110; ">💡 Meta Data</h2> 
<table>    
 <tr>         
	<th >              
	     <p style="text-align: right">Title: </p>          <!-- 标题 -->
	</th>         
	<td >${topItem.getField('title')}         
	</td>      
 </tr> 
 <tr>
        <th >             
	   <p style="text-align: right">Journal: </p>        <!-- 期刊  -->
       </th>         
       <td >${topItem.getField('publicationTitle')}   
       </td>    
 </tr>     
 <tr>
        <th >
            <!-- 作者 -->
            <p style="text-align: right">Authors: </p>
        </th>
        <td >${topItem.getCreators().map((v)=>v.lastName+" "+v.firstName).join("; ")}
        </td>
 </tr>
 <tr>
        <th >   <!-- 出版日期 -->
            <p style="text-align: right">Pub.date: </p>
        </th>
        <td >${topItem.getField('date')}
        </td>
 </tr>   
 <tr>
        <th >
            <p style="text-align: right">DOI: </p>
        </th>
        <td >
            <a href= "https://doi.org/${topItem.getField('DOI')}">${topItem.getField('DOI')}</a>
        </td>
 </tr>
 <tr>
        <th 
            <!-- 期刊标签 -->
            <p style="text-align: right">期刊标签: </p>
        </th>
        <td >${topItem.getField('seriesText')}
        </td>
 </tr>
  <tr>
        <th >
            <!-- 分区 -->
            <p style="text-align: right">分区: </p>
        </th>
        <td >${topItem.getField('seriesTitle')}
        </td>
 </tr>
 <tr>
        <th >
             <!-- 影响因子 -->
            <p style="text-align: right">IF: </p>
        </th>
        <td >${topItem.getField('callNumber')}
        </td>
 </tr>
 <tr>
        <th >
             <!-- 影响因子 -->
            <p style="text-align: right">IF(5): </p>
        </th>
        <td >${topItem.getField('rights')}
        </td>
 </tr>
 </table>
</span></h2>\n <h2 style="color:  #1E5110; ">📜一.研究背景</h2>
<h4>  <span style="color: #1565C0">1.1 背景及相关研究：</span></h4>
<p></p>
<h4>  <span style="color: #1565C0">1.2 解决的问题及所用的方法：</span></h4>
<p></p>
<h4>  <span style="color: #1565C0">1.3 关键技术：</span></h4>
<p></p>
<h4>  <span style="color: #1565C0">1.4 本文主要工作：</span></h4>
<p></p>
</span></h2>\n <h2 style="color:#1E5110; ">📕二.研究内容</h2>
<p></p>
</span></h2>\n <h2 style="color:#1E5110; ">🚩三.研究结论</h2> 
<p></p>
</span></h2>\n <h2 style="color: #1E5110; ">📌四.创新 & 疑问</h2>
<h4>  <span style="color: #1565C0">4.1 创新点：</span></h4>
<p></p>
<h4>  <span style="color: #1565C0">4.2 不足点：</span></h4>
<p></p>
<h4>  <span style="color: #1565C0">4.3 疑问：</span></h4>
<p></p>
</span></h2>\n <h2 style="color:#1E5110; ">🔬五.研究展望</h2>
<p></p>
</span></h2>\n <h2 style="color:#1E5110; ">🖍️六.知识扩充</h2>
<h4>  <span style="color: #1565C0">6.1 专业名词：</span></h4>
<p></p>
<h4>  <span style="color: #1565C0">6.2 相关知识点：</span></h4>
<p></p>

```

效果如下：


![2023-05-11_160503](https://github.com/windingwind/zotero-better-notes/assets/52536738/c7eeb943-6b27-42ac-8638-f038ab997ea8)

将笔记导出为Markdown格式在Obsidian中展示，其效果如下：

![2023-05-11_161709](https://github.com/windingwind/zotero-better-notes/assets/52536738/ce2226f8-f3d8-40ee-a8f6-e758f483768f)

**Reply by zhouying2023** [reply]
想问一下，可以插入meta信息，但是我的期刊标签、IF全部没有内容。我的参考文献列表中通过style和easyscholar是可以呈现这些信息的。请问有可以解决的办法吗

**Reply by Wang-Xiaofei** [reply]
有的，这是我的解决方案：

**所需插件**：Zotero更新影响因子(https://github.com/redleafnew/zotero-updateifs)

- 1）首先，安装上面这个插件；
- 2）然后，配置信息（其实是查看信息的存储位置）。在Zotero->工具->更新期刊信息位置中查看，如下图；相关信息也会在Zotero的文献信息栏中体现；
![Snipaste_47](https://github.com/windingwind/zotero-better-notes/assets/52536738/3ad86699-e7f7-4d82-82e2-0027d1198b9e)
- 3）最后，修改模板中的相关信息。例如：期刊标签中的`seriesText`；及分区中的`seriesTitle`（注意：修改模板中的信息时，要写存储位置的英文名称（Zotero英文版名称），例如中科院的大小分区我存到了`系列标题`，在模板Mete的分区中要改为`seriesTitle`；）

```HTML
<tr>
        <th 
            <!-- 期刊标签 -->
            <p style="text-align: right">期刊标签: </p>
        </th>
        <td >${topItem.getField('seriesText')}
        </td>
 </tr>
  <tr>
        <th >
            <!-- 分区 -->
            <p style="text-align: right">分区: </p>
        </th>
        <td >${topItem.getField('seriesTitle')}
        </td>
 </tr>
 <tr>
        <th >
             <!-- 影响因子 -->
            <p style="text-align: right">IF: </p>
        </th>
        <td >${topItem.getField('callNumber')}
        </td>
 </tr>
 <tr>
        <th >
             <!-- 影响因子 -->
            <p style="text-align: right">IF(5): </p>
        </th>
        <td >${topItem.getField('rights')}
        </td>
 </tr>
```

如果不想麻烦，配置信息直接按我的改就行，模板不用改！

**Reply by zhouying2023** [reply]
感谢，已经解决！

**Reply by muyuan-ke** [reply]
大大，按这个模板总出现ReferenceError:topltem is not defined，是什么情况呀

**Comment by ghweili** [comment]
请问这个meta信息可以也用yaml的形式给出吗，这样导出到obsidian时还可以用于检索，比如按年份、topic、作者列表、期刊、标签（如#from_zotero 等）

**Reply by Wang-Xiaofei** [reply]
如果是使用Obsidian的dataview插件检索的话，我的解决方案是这样的：

1. 将上述模板中的Meta信息`Title：`改为`Title::`(注意：`::`为英文输入法下的符号)
2. 导入到obsidian后，类似这样：![Snipaste10](https://github.com/windingwind/zotero-better-notes/assets/52536738/6e1cc7df-f06a-415a-9c4a-f62c7af02813)
3. 使用dataview插件检索后的形式，类似这样：![Snipaste2](https://github.com/windingwind/zotero-better-notes/assets/52536738/f8ac9e06-c813-4356-a157-aa462c4b0a69)

不过有一点不足的是：这样在dataview插件检索后的结果中会多出两个`|`

**Comment by keji1115** [comment]
Template [item]MeteData table Error: ReferenceError: topItem is not defined
请问会显示上述内容是什么原因？

**Reply by cyinen** [reply]
模板名字修改一下就可以
https://github.com/windingwind/zotero-better-notes/issues/553#:~:text=OK%2C%20I%20figured%20it%20out%20%2D%20the%20template%20should%20actually%20be%20named%20%22%5BItem%5D%20...%20%22%20..%20then%20it%20works%3A
